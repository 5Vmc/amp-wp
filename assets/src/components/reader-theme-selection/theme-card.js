
/**
 * WordPress dependencies
 */
import { useContext } from '@wordpress/element';
import { decodeEntities } from '@wordpress/html-entities';
import { __ } from '@wordpress/i18n';

/**
 * External dependencies
 */
import PropTypes from 'prop-types';

/**
 * Internal dependencies
 */
import { Options } from '../options-context-provider';
import { Selectable } from '../selectable';
import { Phone } from '../phone';
import { AMPNotice, NOTICE_TYPE_WARNING, NOTICE_SIZE_SMALL } from '../amp-notice';

/**
 * A selectable card showing a theme in a list of themes.
 *
 * @param {Object} props Component props.
 * @param {string} props.description Theme description.
 * @param {string} props.currentlyActiveThemeNotice The notice to show if the theme is unselectable.
 * @param {string} props.homepage Link to view more information about the theme.
 * @param {string} props.screenshotUrl URL for screenshot of theme.
 * @param {string} props.slug Theme slug.
 * @param {string} props.name Theme name.
 * @param {boolean} props.disabled Whether the theme is not automatically installable in the current environment.
 */
export function ThemeCard( { currentlyActiveThemeNotice, description, homepage, screenshotUrl, slug, name, disabled } ) {
	const { editedOptions, updateOptions } = useContext( Options );
	const { reader_theme: readerTheme } = editedOptions;

	const id = `theme-card__${ slug }`;

	return (
		<Selectable className={ `theme-card` } direction="bottom" elementName="li" selected={ readerTheme === slug }>
			<label htmlFor={ id } className="theme-card__label">
				<Phone>
					<img
						src={ screenshotUrl }
						alt={ name }
						height="2165"
						width="1000"
						loading="lazy"
						decoding="async"
					/>
				</Phone>
				<div className="theme-card__label-header">
					<input
						disabled={ Boolean( disabled ) }
						type="radio"
						id={ id }
						checked={ readerTheme === slug }
						onChange={ ( ) => {
							updateOptions( { reader_theme: slug } );
						} }
					/>
					<h3>
						{ decodeEntities( name ) }
					</h3>

				</div>

				{ disabled && currentlyActiveThemeNotice && (
					<AMPNotice type={ NOTICE_TYPE_WARNING } size={ NOTICE_SIZE_SMALL }>
						{ currentlyActiveThemeNotice }
					</AMPNotice>
				) }

				<p className="theme-card__description">
					{ decodeEntities( description ) }

				</p>
			</label>
			<p className="theme-card__theme-link">
				<a href={ homepage } target="_blank" rel="noreferrer noopener">
					{ __( 'Learn more', 'amp' ) }
				</a>
			</p>
		</Selectable>
	);
}

ThemeCard.propTypes = {
	currentlyActiveThemeNotice: PropTypes.string,
	description: PropTypes.string.isRequired,
	homepage: PropTypes.string.isRequired,
	screenshotUrl: PropTypes.string.isRequired,
	slug: PropTypes.string.isRequired,
	name: PropTypes.string.isRequired,
	disabled: PropTypes.bool,
};
