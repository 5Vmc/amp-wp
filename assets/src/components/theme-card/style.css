.theme-card {
	display: flex;
	flex-direction: column;
	flex-shrink: 0;
	padding: 1.5rem;
	position: relative;
}

.theme-card p {
	font-size: 14px;
}

.theme-card--disabled {
	position: relative;
}

.theme-card--disabled::before {
	content: "";
	position: absolute;
	top: 0;
	right: 0;
	bottom: 0;
	left: 0;
	background-color: rgba(0, 0, 0, 0.05);
	pointer-events: none;
	z-index: 3;
}

.theme-card__label-header {
	align-items: center;
	display: flex;
	padding: 0.75rem 0;
}

.theme-card__label-header > input {
	flex-shrink: 0;
}

.theme-card__label-header .h3 {
	line-height: 1.3;
	font-size: 1.25rem;
	margin-left: 5px;
}

.theme-card .phone-inner {
	max-height: 307px; /* Make all images uniform height. */
}

.theme-card img {
	height: auto;
	width: 100%;
}

p.theme-card__description {
	font-size: 14px;
	line-height: 1.666;
	display: -webkit-box;
	-webkit-box-orient: vertical;
	-webkit-line-clamp: 3;
	overflow: hidden;
}

.theme-card__theme-link {
	margin-top: auto;
}

.theme-card__disabled-overlay {
	position: absolute;
	top: 0;
	right: 0;
	bottom: 0;
	left: 0;
	display: flex;
	align-items: center;
	justify-content: center;
	background: rgba(255, 255, 255, 0.9);
	font-weight: 700;
	font-size: 1.122rem;
}
