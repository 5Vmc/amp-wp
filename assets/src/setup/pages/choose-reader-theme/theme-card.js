
/**
 * WordPress dependencies
 */
import { useContext } from '@wordpress/element';
import { decodeEntities } from '@wordpress/html-entities';

/**
 * External dependencies
 */
import PropTypes from 'prop-types';

/**
 * Internal dependencies
 */
import './style.css';
import { __ } from '@wordpress/i18n';
import { Options } from '../../components/options-context-provider';

export function ThemeCard( { description, homepage, screenshotUrl, slug, name } ) {
	const { options, updateOptions } = useContext( Options );
	const { reader_theme: readerTheme } = options || {};

	const id = `theme-card__${ slug }`;

	return (
		<li className={ `amp-wp-theme-card ${ readerTheme === slug ? 'amp-wp-theme-card--selected' : '' }` }>
			<label htmlFor={ id } className="amp-wp-theme-card__label">
				<img
					src={ screenshotUrl }
					alt={ name }
				/>
				<div className="amp-wp-theme-card__label-header">
					<input
						type="radio"
						id={ id }
						checked={ readerTheme === slug }
						onChange={ ( ) => {
							updateOptions( { reader_theme: slug } );
						} }
					/>
					<h2>
						{ decodeEntities( name ) }
					</h2>
				</div>

				<p>
					{ decodeEntities( description ) }
				</p>
			</label>
			<p className="amp-wp-theme-card__theme-link">
				<a href={ homepage } target="_blank" rel="noreferrer">
					{ __( 'Learn more', 'amp' ) }
				</a>
			</p>
		</li>
	);
}

ThemeCard.propTypes = {
	description: PropTypes.string.isRequired,
	homepage: PropTypes.string.isRequired,
	screenshotUrl: PropTypes.string.isRequired,
	slug: PropTypes.string.isRequired,
	name: PropTypes.string.isRequired,
};
