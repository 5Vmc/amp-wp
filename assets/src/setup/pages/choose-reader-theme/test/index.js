
/**
 * External dependencies
 */
import { create } from 'react-test-renderer';

/**
 * Internal dependencies
 */
import { ThemeCard } from '../theme-card';
import { Providers } from '../../..';

let container;

describe( 'ThemeCard', () => {
	beforeEach( () => {
		container = document.createElement( 'div' );
		document.body.appendChild( container );
	} );

	afterEach( () => {
		document.body.removeChild( container );
		container = null;
	} );

	it( 'matches snapshot', () => {
		const wrapper = create(
			<Providers>
				<ThemeCard homepage="http://mysite.com" description="Theme card" screenshotUrl="http://screenshot.jpeg" slug="theme-slug" name="Theme name" />
			</Providers>,
		);
		expect( wrapper.toJSON() ).toMatchSnapshot();
	} );
} );
