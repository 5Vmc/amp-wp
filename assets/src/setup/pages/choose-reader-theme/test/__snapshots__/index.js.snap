// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeCard matches snapshot 1`] = `
<li
  className="amp-wp-theme-card "
>
  <label
    className="amp-wp-theme-card__label"
    htmlFor={0}
  >
    <img
      alt="Theme name"
      src="http://screenshot.jpeg"
    />
    <div
      className="amp-wp-theme-card__label-header"
    >
      <input
        checked={false}
        id={0}
        onChange={[Function]}
        type="radio"
      />
      <h2>
        Theme name
      </h2>
    </div>
    <p>
      Theme card
    </p>
  </label>
  <p
    className="amp-wp-theme-card__theme-link"
  >
    <a
      href="http://mysite.com"
      rel="noreferrer"
      target="_blank"
    >
      Learn more
    </a>
  </p>
</li>
`;
